utils/filesystem/files_test.go

```diff
@@ -2030,3 +2030,261 @@ func TestConvertToOSFile(t *testing.T) {
 	require.NoError(t, osFile.Close())
 	require.Error(t, rFile.Close(), "file must already be closed")
 }
+
+func createTestFileStructure(vfs afero.Fs, rootDir string) error {
```

Comment:
you should use the filesystem representations from this module instead of afero.Fs

Although I don't think you need to use a different filesystem at all, just use GetGlobalFileSystem() since you are writing to the filesystem.

This might make more sense if you were testing the in memory file system as well